### PR TITLE
XIVY-11246 Fix npe after user switch if orignal url is case/task detail

### PR DESCRIPTION
**Pull request by ivy-lli**
The problem is not that the case/task id is not on the parameter, the problem is that if you do a redirect to the login page, then a POST request is first triggered (without any query params). The original URL is then evaluated from this POST request instead of the inital page request.

### CODE REVIEW

File: dev-workflow-ui/webContent/includes/template/template.xhtml [hunk 1]
```diff
@@ -56,16 +56,16 @@
...
-                  <p:commandLink id="loginTableBtn" actionListener="#{loginIvyDevWfBean.redirectToLoginTable}" rendered="#{engineModeIvyDevWfBean.demoOrDesigner}">
+                  <p:link id="loginTableBtn" href="loginTable.xhtml?originalUrl=#{loginIvyDevWfBean.originalPage}" rendered="#{engineModeIvyDevWfBean.demoOrDesigner}">
                     <i class="si si-lg si-multiple-neutral-1"></i>
                     <span>Switch user</span>
-                  </p:commandLink>
+                  </p:link>
                 </li>
                 <li>
-                  <p:commandLink id="loginFormBtn" actionListener="#{loginIvyDevWfBean.redirectToLoginForm}">
+                  <p:link id="loginFormBtn" href="login.xhtml?originalUrl=#{loginIvyDevWfBean.originalPage}">
```
Comment (alexsuter) [inline dev-workflow-ui/webContent/includes/template/template.xhtml:65]: which means that this originalUrl was not correct, right?
would be easy to test that this originaUrl is correctly set?

**Review by alexsuter — APPROVED** [review]
Good! Thx!

We should add a test for this

File: dev-workflow-ui-web-test/src_test/ch/ivyteam/ivy/project/workflow/webtest/WebTestLoginIT.java [hunk 2]
```diff
@@ -92,15 +103,29 @@ void testRedirectToOriginalUrl() {
...
+    loginFromTable("testuser");
+    startTestProcess("1750C5211D94569D/TestData.ivp");
+    openView("allTasks.xhtml");
+    $(".detail-btn").shouldBe(visible).click();
+    var taskId = $(By.id("taskId")).shouldBe(visible).text();
+    assertCurrentUrlContains("taskDetails.xhtml?task=" + taskId);
+
+    $(".user-profile").shouldBe(visible).click();
+    $(By.id("loginTableBtn")).shouldBe(visible).click();
+    assertCurrentUrlContains("loginTable.xhtml?originalUrl=taskDetails.xhtml%3Ftask%3D" + taskId);
```
Comment (ivy-lli) [inline dev-workflow-ui-web-test/src_test/ch/ivyteam/ivy/project/workflow/webtest/WebTestLoginIT.java:116]: Test that the originalUrl now contains also the taskid

File: dev-workflow-ui-web-test/src_test/ch/ivyteam/ivy/project/workflow/webtest/util/WorkflowUiUtil.java [hunk 3]
```diff
@@ -55,31 +55,30 @@ private static String viewUrl(String page) {
   }
 
   public static void assertCurrentUrlContains(String contains) {
-    WebDriverConditions.currentFrameUrlContaining(contains);
```
Comment (ivy-lli) [inline dev-workflow-ui-web-test/src_test/ch/ivyteam/ivy/project/workflow/webtest/util/WorkflowUiUtil.java:58]: This never does any assertion, this is only a condition.
Fixing this has broken over half of all tests...

**Pull request merged**